### PR TITLE
#1662 Prescriptions Wizard

### LINKED ISSUE #1662: Presciption wizard page
## Is your feature request related to a problem? Please describe.

Need an initial start on the prescription wizard page for dispensing.

## Describe the solution you'd like

As per the wire frame on balsamiq

## Implementation

Should be a basic iteration which has basic functionality - most likely to choose a prescriber, an item and have a summary

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
### BRANCHED FROM #1630

Fixes #1662 

## Change summary

- Adds the basic structure for the prescriptions Wizard for to enable more focused iterations
- Adds a `PrescriptionReducer`  - Not sure on this. I think having a reducer and state for this flow is good and there will many more actions/reducer cases/state to hold when there is full functionality, but not sure if it should be split up more. For example, a `TabsReducer` which handles the state of the tabs, which would be re-used elsewhere. Also an `ItemsReducer` which handles interactions with the items. a `PaymentsReducer` which handles payments. Seems over the top right now, but possibly this would be the way to go.
- Couple other minor offroad adventures: added a star icon, added forwarding of a ref to `SimpleTable` to control the scrolling, changed a few propTypes in order to handle new use cases

prescriptionwizard1

- Also not sure what to do with the many-components-makeup of the page. Maybe `WizardTabs` folder is a good idea but not sure

## Testing

N/A

### Related areas to think about

I prefer to have smaller issues, so trying to get this main structure part done and in and iterate on that with more focused code changes. Next steps would be:

- Add quantity increment/decrement feature changes
- Add finalise logic
- Add abbreviation/direction feature
- Add payment features
- "Favourites" feature
- Tidy up UI [broken up further]
- Performance enhancements

### CODE REVIEW

File: src/pages/PrescriptionPage.js [hunk 1]
```diff
@@ -4,213 +4,189 @@
...
+import { getColumns } from './dataTableUtilities';
+
+import { selectItem, selectPrescriber, switchTab } from '../reducers/PrescriptionReducer';
+
+/**
+ * File contains Four components for the PrescriptionPage. The container component Prescription and
+ * three "Tab" components PrescriberSelect/ItemSelect/Summary.
+ */
+
+const ALPHABET = [...'abcdefghijklmnopqrstuvwxyz'];
```
Comment (mark-prins) [inline src/pages/PrescriptionPage.js]: perhaps move to a constants file?
We talked about localisation at one point and decided that the latin character set is fine, right? We'll have a look at localised versions later..

File: src/pages/PrescriptionPage.js [hunk 2]
```diff
@@ -4,213 +4,189 @@
...
+
+const mapDispatchToProps = dispatch => {
+  const choosePrescriber = prescriberID => dispatch(selectPrescriber(prescriberID));
+  const chooseItem = itemID => dispatch(selectItem(itemID));
+  const changeTab = currentTab => dispatch(switchTab(currentTab + 1));
```
Comment (mark-prins) [inline src/pages/PrescriptionPage.js]: more precisely this is `nextTab`

File: src/pages/PrescriptionPage.js [hunk 3]
```diff
@@ -4,213 +4,189 @@
...
-const mapStateToProps = state => {
-  const { pages } = state;
-  const { prescription } = pages;
+});
 
-  return prescription;
+const Summary = connect(mapStateToProps)(({ transaction }) => (
+  <View style={{ flex: 1 }}>
+    {transaction.items.map(item => (
+      <View>
```
Comment (mark-prins) [inline src/pages/PrescriptionPage.js]: doesn't this need a `key` prop?

File: src/reducers/PrescriptionReducer.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+};
+
+export const switchTab = nextTab => ({
+  type: ACTIONS.SWITCH_TAB,
+  payload: { nextTab },
+});
+
+export const selectPrescriber = prescriberID => (dispatch, getState) => {
+  const { prescription } = getState();
+
```
Comment (mark-prins) [inline src/reducers/PrescriptionReducer.js]: There's

A lot of

Line breaks here.

File: src/widgets/SimpleTable.js [hunk 5]
```diff
@@ -20,90 +21,100 @@ import { HeaderCell, HeaderRow } from './DataTable/index';
...
+  })
+);
+
+SimpleTable.defaultProps = {
+  extraData: {},
+};
 
 SimpleTable.propTypes = {
-  data: PropTypes.array.isRequired,
+  data: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
```
Comment (mark-prins) [inline src/widgets/SimpleTable.js]: The page comment still refers to `data` as an array only

Comment (mark-prins) [reply src/widgets/SimpleTable.js]: And `FlatList` says that it takes a `ReadOnlyArray` for the `data` property - presumably an object is read as an array of properties and just works?

File: src/widgets/Stepper.js [hunk 6]
```diff
@@ -66,7 +66,7 @@ const StepperNumber = ({ step, numberOfSteps, currentStep, title, onPress }) =>
         {title}
       </Text>
 
-      {lastStep && <View style={stepSeperator} />}
+      {!lastStep && <View style={stepSeperator} />}
```
Comment (mark-prins) [inline src/widgets/Stepper.js]: oh! oops, I must have missed that!

**Comment by Chris-Petty** [comment]
I'm startled by the amount of screen realestate you have in gifs. Might be worthwhile trying to set the emulator res to 1024x768 (😢)

File: src/pages/DispensingPage.js [hunk 7]
```diff
@@ -12,7 +12,8 @@ import { ToggleBar, DataTablePageView, SearchBar, PageButton } from '../widgets'
 import { DataTable, DataTableRow, DataTableHeaderRow } from '../widgets/DataTable';
 
 import globalStyles from '../globalStyles';
-import { PageActions, DATA_SET, getPageDispatchers, getItemLayout } from './dataTableUtilities';
+import { PageActions, DATA_SET, getItemLayout, getPageDispatchers } from './dataTableUtilities';
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/pages/DispensingPage.js]: ..alphabetical order? wait..

File: src/pages/PrescriptionPage.js (hunk 1 above)
Comment (joshxg) [reply src/pages/PrescriptionPage.js]: honestly, I couldn't decide where to put the constants folder so I didn't make one. I've put it in `src/widgets/constants.js` - not sure 🤷‍♂ 

ff5c937e5c51cbd8bdf7ce2cbbaca2352663456c

File: src/pages/PrescriptionPage.js (hunk 2 above)
Comment (joshxg) [reply src/pages/PrescriptionPage.js]: fe842f9cbb02308089e40acf665b851f4df9c542

File: src/pages/PrescriptionPage.js (hunk 3 above)
Comment (joshxg) [reply src/pages/PrescriptionPage.js]: ae8bc54918ac9b217f095692f2dba48f7c6e1739

File: src/reducers/PrescriptionReducer.js (hunk 4 above)
Comment (joshxg) [reply src/reducers/PrescriptionReducer.js]: 0b8fdd389274ceaa8dffd858fa70f5de63172216

File: src/widgets/SimpleTable.js (hunk 5 above)
Comment (joshxg) [reply src/widgets/SimpleTable.js]: Tried to clarify here: f89f49ad74a85c1d4ef2db760270bf7042947c40

A "realm array" or `Realm.results` is an object `{ 0, 1, 2, .. n }` so is an array that works.. can we get rid of realm yet

Comment (mark-prins) [reply src/widgets/SimpleTable.js]: oh. sorry I asked

**Pull request merged**